contracts/src/LightClient.sol

```diff
@@ -9,16 +16,24 @@ import { LightClientStateUpdateVK as VkLib } from "./libraries/LightClientStateU
 
 /// @notice A light client for HotShot consensus. Keeping track of its finalized states in safe,
 /// authenticated ways.
-contract LightClient {
+contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     // === Events ===
     //
     // @notice Notify a new epoch is starting
     event EpochChanged(uint64);
 
+    /// @notice upgrade event when the proxy updates the implementation it's pointing to
+    event Upgrade(address implementation);
+
     // === Constants ===
     //
     /// @notice System parameter: number of blocks per epoch
-    uint32 public immutable BLOCKS_PER_EPOCH;
+    /// @dev This variable cannot be made immutable due to how UUPS contracts work. See
+    /// https://forum.openzeppelin.com/t/upgradable-contracts-instantiating-an-immutable-value/28763/2#why-cant-i-use-immutable-variables-1
+    uint32 public blocksPerEpoch;
+
+    /// @notice A simple way to track contract versions
+    uint32 public immutable VERSION = 1;
```

Comment:
should we adopt major, minor, patch semver like [this one](https://github.com/ethereum-optimism/optimism/blob/v1.1.4/packages/contracts-bedrock/src/universal/Semver.sol)?

Reply:
Good idea. Done in aac21637b0aeb05bc3c0f33fbe375206e0043551. Tell me if it is what you had in mind and I will apply it to the Fee contract. Note that using a Semver contract does not quite work because it messes up with the order of the variables and also one cannot use the constructor.